src/jit/morph.cpp

```diff
@@ -104,6 +99,20 @@ GenTreePtr Compiler::fgMorphIntoHelperCall(GenTreePtr tree, int helper, GenTreeA
     }
 #endif // _TARGET_XXX_
 
+    if (tree->OperMayThrow(this))
+    {
+        tree->gtFlags |= GTF_EXCEPT;
```

Comment:
Maybe we should add some flags setters to `GenTree`, this stuff keeps repeating.

`tree->SetFlag(GTF_EXCEPT, tree->OperMayThrow(this))` for example.